include/revng/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,1083 @@
+  /// the GPRs if `MaximumGPRsPerAggregateArgument` is less than or equal to
+  /// the number of the registers required to fit the object including padding.
+  ///
+  /// \note: If `MaximumGPRsPerAggregateArgument` is equal to 0, it means that
+  /// the ABI does not allow aggregate arguments to use GPRs.
+  ///
+  /// \note: If an argument doesn't fit into the specified registers or
+  /// uses irregular padding, the registers are not used and the object is
+  /// passed using the memory (stack, pointer-to-copy, etc.).
+  static constexpr size_t MaximumGPRsPerAggregateArgument = 0;
+
+  /// States the maximum number of GPRs available to return a single aggregate
+  /// (a struct, a union, etc.) value, meaning that it can only be returned
+  /// in the GPRs if `MaximumGPRsPerAggregateReturnValue` is less than or equal
+  /// to the number of the registers required to fit the object including
+  /// padding.
+  ///
+  /// \note: If `MaximumGPRsPerAggregateReturnValue` is equal to 0, it means
+  /// that the ABI does not allow aggregate return values to use GPRs.
+  ///
+  /// \note: If a return value doesn't fit into the specified registers or
+  /// uses irregular padding, the registers are not used and the object is
+  /// passed using the memory (stack, pointer-to-copy, etc.).
+  static constexpr size_t MaximumGPRsPerAggregateReturnValue = 0;
+
+  /// States the maximum number of GPRs available to pass a single scalar
+  /// (`int`, `__int128`, pointer, etc.) argument, meaning that it can only be
+  /// passed in the GPRs if `MaximumGPRsPerScalarArgument` is less than or
+  /// equal to the number of the registers required to fit the object.
+  ///
+  /// \note: If `MaximumGPRsPerScalarArgument` is equal to 0, it means that
+  /// the ABI does not allow scalar arguments to use GPRs.
+  ///
+  /// \note: If an argument doesn't fit into the specified registers or
+  /// uses irregular padding, the registers are not used and the object is
+  /// passed using the memory (stack, pointer-to-copy, etc.).
+  static constexpr size_t MaximumGPRsPerScalarArgument = 0;
+
+  /// States the maximum number of GPRs available to return a single scalar
+  /// (`int`, `__int128`, pointer, etc.) value, meaning that it can only be
+  /// returned in the GPRs if `MaximumGPRsPerScalarReturnValue` is less than
+  /// or equal to the number of the registers required to fit the object
+  /// including padding.
+  ///
+  /// \note: If `MaximumGPRsPerScalarReturnValue` is equal to 0, it means
+  /// that the ABI does not allow scalar return values to use GPRs.
+  ///
+  /// \note: If a return value doesn't fit into the specified registers or
+  /// uses irregular padding, the registers are not used and the object is
+  /// passed using the memory (stack, pointer-to-copy, etc.).
+  static constexpr size_t MaximumGPRsPerScalarReturnValue = 0;
+
+  /// States the required size for a single stack argument in bytes.
+  static constexpr size_t MinimumStackArgumentSize = 1;
```

Comment:
Why is this important?

Reply:
It could be useful when detecting stack alignment, could it not?

But, yeah, when you see anything stack-related you can consider it heavily-WIP.

Reply:
Didn't you have a pad for things that's useful to keep track of but are not used yet?
Maybe move it there.

Reply:
The pad was for things that don't appear in the actual implementation in any way. This one is actually mentioned when converting CABI->Raw (but now that I look closer at it, the whole thing is obsolete, at least for now.

Either way, I think this will still be important when building `argument`-`stack offset` relation map. I'll still remove it for now, I think.

Reply:
Moved it to the pad.